lib/app.rb

```diff
@@ -161,4 +168,46 @@ def add_book
     puts 'Book created successfully!'
     @books << book
   end
+
+  def list_musicalbum
+    if @musicalbums.length.zero?
+      puts 'No Music Album added yet !'
+    else
+
+      @musicalbums.each_with_index do |album, index|
+        puts "#{index + 1} Music Album :"
+        puts album
+      end
+    end
+  end
+
+  def list_genre
+    if @genres.length.zero?
+      puts 'No Genre registered yet!'
+    else
+
+      @genres.each_with_index do |genre, index|
+        puts "#{index + 1}) Genre : #{genre}"
+      end
+    end
+  end
+
+  def create_musicalbum
+    genre, author, label = retrieve_objects
+    published_date = [(print 'Published date (yyyy-mm-dd): '), gets.rstrip][1]
+    year, month, day = published_date.split('-')
+    on_spotify = [(print 'Is this album on spotify (Type True or False): '), gets.rstrip][1]
+    begin
+      musicalbum = MusicAlbum.new(Date.new(year.to_i, month.to_i, day.to_i), on_spotify)
```

Comment:
Please check that the value of `on_spotify` is a boolean, since that is how you defined the class and is part of the requirements. It can be done by setting a condition statement directly from the input (at the end):
```ruby
on_spotify = [(print 'Is this album on spotify (Type True or False): '), gets.rstrip][1] == "True"
```

And on the input for creating your music album object, remember that booleans with default values need the use of `:` in the parameter like so:

```ruby
musicalbum = MusicAlbum.new(Date.new(year.to_i, month.to_i, day.to_i), on_spotify: on_spotify)
```
That just means
```ruby
parameter: value
```